### DISCUSSION TITLE
Network printer configuration on clients

### DISCUSSION THREAD

**Discussion by diegobenincasa** [ltsp/ltsp#221] [Q&A]
Hello!

I have an environment with 60+ LTSP clients divided into departments, and each department has irs own printer. Every printer on this environment is connected to the network.

Before LTSP, printers were discoverable and easily installed on every desktop. But with LTSP, installed printers are lost between logins, so I have to install them again and again at each reboot.

Setting printers on lts.conf doesn't seem to be the most adequate procedure, as there is a lot of printers on the network and so many clients to set up individually (not the same printers for all the clients).

Is there a way to persist installed printers between reboots? Or I really should install every printer on the server and set them up on a per-client basis?

Thanks in advance!

**Comment by alkisg** [comment]
Check the solution described in https://github.com/ltsp/ltsp/issues/8#issuecomment-554458466.
With this one, the server autodiscovers the printers, and automatically shares them to all the clients. Is it enough?

**Comment by diegobenincasa** [comment]
I'll try that solution and will let you know if that helped or not.

If that works, I guess I have to install every single printer (and that might mean additional drivers) on the server to make them available to anyone who wants to use any available printer, right?

**Comment by alkisg** [comment]
Exactly. The LTSP server then **is** the CUPS server; the LTSP clients don't have their own CUPS server. And that way all printers are managed centrally.

**Comment by diegobenincasa** [comment]
Glad to say to you that it works now. Thanks!

But I have one more question now...

Some printers are actually multifunction printers (all-in-one), so the scanner part is usually controlled by SANE. Does the CUPS procedure also applies to SANE (of course, with its own configuration parameters), i.e, a central server controlling every network scanner? Right now they're not found on the network, even though the printers are working.

If yes, how to make them work?

**Comment by maccadia** [comment]
SANE seems to be able to run in client-server mode over the network : https://serverfault.com/questions/162061/can-i-share-the-scanner-functionality-of-a-multifunction-printer-over-a-network

**Comment by alkisg** [comment]
I invite you to use google translate on [this forum thread](https://alkisg.mysch.gr/steki/index.php?topic=6738.0) and write an LTSP community wiki page about scanner sharing.

The client part would be a single ltsp.conf parameter, e.g.

```shell
POST_INIT_SANE_NET="echo server >> /etc/sane.d/net.conf"
```
